src/nvim/regexp.c

```diff
@@ -7016,12 +7003,22 @@ regprog_T *vim_regcomp(char_u *expr_arg, int re_flags)
             BT_REGEXP_DEBUG_LOG_NAME);
     }
 #endif
-    /*
-     * If the NFA engine failed, the backtracking engine won't work either.
-     *
-       if (regexp_engine == AUTOMATIC_ENGINE)
-        prog = bt_regengine.regcomp(expr, re_flags);
-     */
+    // If the NFA engine failed, try the backtracking engine.
+    // Disabled for now, both engines fail on the same patterns.
+    // Re-enable when regcomp() fails when the pattern would work better
+    // with the other engine.
+    //
+    // if (regexp_engine == AUTOMATIC_ENGINE) {
```

Comment:
Lint won't be happy about commented out code, but it's in the patch and will be fixed in a later patch.